client/allochealth/tracker.go

```diff
@@ -523,14 +529,20 @@ OUTER:
 	}
 }
 
-func evaluateConsulChecks(tg *structs.TaskGroup, registrations *serviceregistration.AllocRegistration) bool {
+func evaluateConsulChecks(
+	alloc *structs.Allocation,
+	tg *structs.TaskGroup,
+	taskEnv *taskenv.TaskEnv,
+	registrations *serviceregistration.AllocRegistration,
+) bool {
```

Comment:
We're now passing a bunch of stuff here. Another option would be to perform the interpolation outside `evaluateConsulChecks` and just pass the list of services, or make this a method of `Tracker`.

Reply:
Discussion covered in https://github.com/hashicorp/nomad/pull/16402#discussion_r1130228868